src/utils/warn.js

```diff
@@ -4,6 +4,13 @@
  */
 /* istanbul ignore next */
 function warn(message) {
+  // eslint-disable-next-line no-undef
+  if (VueBootstrap && VueBootstrap.warnHandler) {
```

Comment:
The proper name is `BoostrapVue`

And this assumes that the name is global (i.e. assigned as `window.BootstrapVue`).

It would be better to pass this as an option when `Vue.use(BootstrapVue, { /*options object */})`

And have the plugin check for the second argument when instantiating (it would need to be defined as a property on the main install routine and bound to `window` or `global`, depending on the environment it is running in (i.e. client or SSR)

Reply:
Another option would be to optionally compile a no-op function depending on DEV or PROD mode of the final project.

Reply:
@tmorehouse Thank you for the amazingly fast review here! 🚀 

> The proper name is BoostrapVue

I found `VueBootstrap.warnHandler` closer to `Vue.config.warnHandler` but also don't mind reversing it to match the project name. 😃 

> It would be better to be able to optionally pass this as an option via Vue.use(BootstrapVue, { /*options object */})

Totally agree—but it does not work for the case where you don't register BootstrapVue components globally, correct?

> depending on DEV or PROD mode of the final project.

you mean like

```javascript
if (process.env.NODE_ENV === 'development') {
  console.warn(`[BootstrapVue warn]: ${message}`)
}
```

?

That still wouldn't allow failing tests for warnings but would be a very helpful first step. 👍

Reply:
The other option would be maybe a minification process... if in the minified version, `no-op` the method out, and in the non-minified version it would warn to the console. (i.e. setting global vars for terser: https://github.com/terser-js/terser#conditional-compilation during minification to remove the call to console.warn, without removing the warn utility)

But is most cases, the user should be fixing any issues that cause errors to be logged out to the console, rather than hiding them ;-)

Reply:
> But is most cases, the user should be fixing any issues that cause errors to be logged out to the console, rather than hiding them ;-)

Completely agree—in our case however we have more than 300 of these warnings and fixing them will take a while. In the meantime, I'd appreciate less output when running tests. 😃

Reply:
Would a

```javascript
if (process.env.BOOTSTRAP_VUE_NO_WARN) {
  return;
}

console.warn(`[BootstrapVue warn]: ${message}`)
```

work for you?

Reply:
Assuming that it can be handled well if `process && process.env` are tested for existence first, as not all users use a bundler to compile their projects (i.e. users using script tags to load BootstrapVue into their browser page).

Reply:
And warning are there for a reason... not correcting them may cause unpredictable results.

Reply:
I have updated this pull request now. I will squash the commits and add docs tomorrow.

> And warning are there for a reason... not correcting them may cause unpredictable results.

Thanks for your concerns! ❤️ As far as I can tell, the warnings only happen in our test environment (due to not waiting for components to be properly setup). While reducing the confidence in our test suite, at least there seem to be no consequences for our users.

In any case: my wish was actually to be able to fail our newly written tests in case warnings happen, so they cannot be ignored. Maybe can get that implemented in a follow-up?

Alternatively we could introduce a `BOOSTRAP_VUE_WARN=ignore` / `BOOSTRAP_VUE_WARN=throw` to solve the problem:

```javascript
const { BOOSTRAP_VUE_WARN } = process && process.env || {};
if (BOOSTRAP_VUE_WARN === 'ignore') {
  return;
}

if (BOOSTRAP_VUE_WARN === 'throw') {
  throw new Error(message);
}

console.warn(`[BootstrapVue warn]: ${message}`)

Reply:
Throw might be a bit extreme in some cases ;-) and the stack trace might not be overly helpful due to some anonymous function.

Reply:
Not sure what happened with my comment here. 🤔 

Let's stick with hiding the warnings then. I will write some docs later today and try to come up with tests.